python_modules/dagster/dagster/_core/definitions/auto_materialize_reason.py

```diff
@@ -0,0 +1,75 @@
+from enum import Enum
+from typing import NamedTuple
+
+
+class AutoMaterializeResult(Enum):
+    """Represents the set of results of the auto-materialize logic.
+
+    MATERIALIZE: The asset should be materialized by a run kicked off on this tick
+    SKIP: The asset should not be materialized by a run kicked off on this tick, because future
+        ticks are expected to materialize it.
+    DISCARD: The asset should not be materialized by a run kicked off on this tick, but future
+        ticks are not expected to materialize it.
+    """
+
+    MATERIALIZE = "MATERIALIZE"
+    SKIP = "SKIP"
+    DISCARD = "DISCARD"
+
+
+class AutoMaterializeCondition(Enum):
+    """Represents the set of conditions that can impact auto-materialization for an asset."""
+
+    def __init__(self, id: str, result: AutoMaterializeResult):
+        self.id = id
+        self.result = result
+
+    # Materialization Reasons
```

Comment:
Is this use of "Reason" congruous with the name `AutoMaterializeConditionReason`?

Reply:
should be `"Materialize Conditions"`